File: _includes/app/basic-sample.cpp
// Build with g++ -std=c++11 basic-sample.cpp -lpq -lpqxx

#include <cassert>
#include <functional>
#include <iostream>
#include <stdexcept>
#include <string>
#include <pqxx/pqxx>

using namespace std;

int main() {
  try {
    pqxx::connection c("postgresql://maxroach@localhost:26257/bank");

    pqxx::nontransaction w(c);

    w.exec("INSERT INTO accounts (id, balance) VALUES (1, 1000), (2, 230)");

    cout << "Account balances:" << endl;
    pqxx::result r = w.exec("SELECT id, balance FROM accounts");
    for (auto row : r) {
      cout << row[0].as<int>() << ' ' << row[1].as<int>() << endl;
    }

    w.commit();  // Note this doesn't doesn't do anything
                 // for a nontransaction, but is still required.
  }
  catch (const exception &e) {
    cerr << e.what() << endl;
    return 1;
  }
  cout << "Success" << endl;
  return 0;
}

File: _includes/app/basic-sample.php
<?php
try {
  $dbh = new PDO('pgsql:host=localhost;port=26257;dbname=bank;sslmode=disable',
    'maxroach', null, array(PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION));

  $dbh->exec('INSERT INTO accounts (id, balance) VALUES (1, 1000), (2, 250)');

  print "Account balances:\n";
  foreach ($dbh->query('SELECT id, balance FROM accounts') as $row) {
      print $row['id'] . ': ' . $row['balance'] . "\n";
  }
} catch (Exception $e) {
	print $e->getMessage();
	exit(1);
}
?>
